### PR TITLE
Remove name from join token

### PR DESCRIPTION
**Pull request by MggMuggins**
Resolves the titular concern in #96 

@masnax I had mentioned that this caused cluster join to fail and... I can't reproduce it. Cluster forms up nicely. Likely had a messy worktree at the end of the day or some such. This passes what tests there are in #110.

### CODE REVIEW

**Review by roosterfish — APPROVED** [review]
Looks good to me. 

I am wondering if we should hold this back until we have agreed on how to proceed with the join process? (https://discourse.ubuntu.com/t/secure-cluster-join-in-microcloud/44261/18). 

Overall I see it unrelated to the join process as it really is just a label right now. @masnax any thoughts on this?

**Comment by tomponline** [comment]
@MggMuggins please can you see if LXD should also have the node name removed from its join token if its also not used? Thanks

**Comment by tomponline** [comment]
> Looks good to me.
> 
> I am wondering if we should hold this back until we have agreed on how to proceed with the join process? (https://discourse.ubuntu.com/t/secure-cluster-join-in-microcloud/44261/18).
> 
> Overall I see it unrelated to the join process as it really is just a label right now. @masnax any thoughts on this?

Although I don't see an issue with merging this, I do also think we shouldn't close the related issue as the mitm scenario still exists and this change doesn't fix it.

**Comment by MggMuggins** [comment]
@tomponline It looks like the cluster name is being sent from the new cluster joiner [here](https://github.com/canonical/lxd/blob/main/lxd/main_init.go#L168) and is required by the local LXD's API [here](https://github.com/canonical/lxd/blob/main/lxd/api_cluster.go#L338). It looks like the name is used for a couple of additional requests to `/internal` endpoints although my understanding of what goes on during the cluster join process gets fuzzy after the request is submitted to the local LXD.

TL;DR it doesn't look like it will be trivial to remove in LXD like it was here.
